// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package google.crypto.tink;

option java_package = "com.google.crypto.tink.proto";
option java_multiple_files = true;
option go_package = "github.com/tink-crypto/tink-go/v2/proto/kms_aead_go_proto";

message KmsAeadKeyFormat {
  // Required.
  // The location of a KMS key.
  // With Google Cloud KMS, valid values have this format:
  // gcp-kms://projects/*/locations/*/keyRings/*/cryptoKeys/*.
  // With AWS KMS, valid values have this format:
  // aws-kms://arn:aws:kms:<region>:<account-id>:key/<key-id>
  string key_uri = 1;
}

// There is no actual key material in the key.
message KmsAeadKey {
  uint32 version = 1;
  // The key format also contains the params.
  KmsAeadKeyFormat params = 2;
}
